Optional skip_on_field_errors arg for validates_schema

d-sutherland opened the pull request:
Add 'skip_on_field_errors' parameter to validates_schema.
'skip_on_field_errors' allows schema level validation to be skipped
when errors are detected in field validation removing the need for
the user to check.
Address issue #323
Suggested that to further address issue #323 the next major release
should change the default value for this parameter from False to True.

d-sutherland committed:
Optional skip_on_field_errors arg for validates_schema

Add 'skip_on_field_errors' parameter to validates_schema.
'skip_on_field_errors' allows schema level validation to be skipped
when errors are detected in field validation removing the need for
the user to check.
Address issue #323

sloria commented on tests/test_decorators.py:
```diff
@@ -429,3 +429,40 @@ def strict_fields(self, data, original_data):
+
+        class MySchema(Schema):
+            foo = fields.Int(required=True)
+            bar = fields.Int(required=True)
```
Could you also test that the schema validators aren't run if a field validator (i.e. a validator callable passed as the `validate` argument) fails?

sloria commented on marshmallow/schema.py:
```diff
@@ -797,12 +800,17 @@ def _invoke_field_validators(self, data, many):
                     )
 
-    def _invoke_validators(self, pass_many, data, original_data, many):
+    def _invoke_validators(self, pass_many, data, original_data, many, field_errors):
```
In order to maintain backwards-compat, could you make `field_errors` an optional argument?

sloria commented:
This is looking good. Just a few minor changes suggested above, and this should be good to merge. Thanks for the PR!

d-sutherland committed:
Updates to optional skip_on_field_errors arg
Add testing for validate kwarg of fields and make
field_errors argument to invoke_validators optional for
backwards compatibility.
Address issue #323

d-sutherland commented:
Thanks for the comments. Updates have been made.

sloria commented:
Thanks!
